src/main/java/com/o19s/es/ltr/feature/store/ExtraLoggingSupplier.java

```diff
@@ -0,0 +1,28 @@
+package com.o19s.es.ltr.feature.store;
+
+import com.o19s.es.ltr.ranker.LogLtrRanker;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+
+public class ExtraLoggingSupplier implements Supplier<Map<String,Object>> {
```

Comment:
I'm not sure I will be able to use the MutableSupplier. Since the Map is not created until first access, I will need to store an internal Supplier object rather than a Map object.